src/Adapter/Discount/Repository/DiscountRepository.php

```diff
@@ -138,6 +138,25 @@ public function getProductRulesGroup(DiscountId $discountId): array
         return $productRulesGroups;
     }
 
+    public function getIdByCode(string $code): ?int
+    {
+        $code = mb_strtoupper(trim($code));
+
+        $cartRuleId = $this->connection->createQueryBuilder()
+            ->select('id_cart_rule')
+            ->from($this->dbPrefix . 'cart_rule')
+            ->where('code = :code')
```

Comment:
If you uppercase the input you should also uppercase the value in DB in case it was recorded differently

Reply:
```suggestion
            ->where('UPPER (code) = :code')
```
Something like this although, I don't know if it's doable natively with DQL or if we'd need to add some extension